### PR TITLE
Heatmap: Fix null options migration

### PR DESCRIPTION
**Pull request by overvenus**
Some dashboard generation libraries, such as grafanalib, may produce heatmaps with a null 'options' attribute. This condition triggers a panic during migration, resulting in a blank heatmap panel.

This commit addresses the issue by returning default options if a panel has a null 'options'.

---

Exception stack (grafana v9.2.4)
```
Uncaught (in promise) TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at z.onPanelMigration (migrations.ts:15:14)
    at x.pluginLoaded (PanelModel.ts:421:31)
    at actions.ts:34:13
```

Blank heatmap
image

<details><summary>Panel JSON</summary>
<p>

```json
{
  "cacheTimeout": null,
  "cards": {
    "cardPadding": null,
    "cardRound": null
  },
  "color": {
    "cardColor": "#b4ff00",
    "colorScale": "sqrt",
    "colorScheme": "interpolateSpectral",
    "exponent": 0.5,
    "max": null,
    "min": null,
    "mode": "spectrum"
  },
  "dataFormat": "tsbuckets",
  "datasource": "${DS_TEST-CLUSTER}",
  "description": "The time consumed for peer processes to be ready in Raft",
  "editable": true,
  "error": false,
  "fieldConfig": {
    "defaults": {
      "thresholds": {
        "mode": "absolute",
        "steps": []
      }
    }
  },
  "gridPos": {
    "h": 7,
    "w": 12,
    "x": 0,
    "y": 0
  },
  "heatmap": {},
  "height": null,
  "hideTimeOverride": false,
  "hideZeroBuckets": true,
  "highlightCards": true,
  "id": 111,
  "interval": null,
  "legend": {
    "show": false
  },
  "links": [],
  "maxDataPoints": 512,
  "maxPerRow": null,
  "minSpan": null,
  "options": null,
  "repeat": null,
  "repeatDirection": null,
  "reverseYBuckets": false,
  "span": null,
  "targets": [
    {
      "datasource": "${DS_TEST-CLUSTER}",
      "expr": "sum(rate(\n    tikv_raftstore_raft_process_duration_secs_bucket\n    {k8s_cluster=\"$k8s_cluster\",tidb_cluster=\"$tidb_cluster\",instance=~\"$instance\",type=\"ready\"}\n    [$__rate_interval]\n)) by (le) ",
      "format": "heatmap",
      "hide": false,
      "instant": false,
      "interval": "",
      "intervalFactor": 1,
      "legendFormat": "{{le}}",
      "metric": "",
      "query": "sum(rate(\n    tikv_raftstore_raft_process_duration_secs_bucket\n    {k8s_cluster=\"$k8s_cluster\",tidb_cluster=\"$tidb_cluster\",instance=~\"$instance\",type=\"ready\"}\n    [$__rate_interval]\n)) by (le) ",
      "refId": "",
      "step": 10,
      "target": ""
    }
  ],
  "timeFrom": null,
  "timeShift": null,
  "title": "Process ready duration",
  "tooltip": {
    "msResolution": true,
    "shared": true,
    "showHistogram": true,
    "sort": 0,
    "value_type": "individual"
  },
  "transformations": [],
  "transparent": false,
  "type": "heatmap",
  "xAxis": {
    "mode": "time",
    "name": null,
    "show": true,
    "values": []
  },
  "xBucketNumber": null,
  "xBucketSize": null,
  "yAxis": {
    "decimals": 1,
    "format": "s",
    "label": null,
    "logBase": 1,
    "max": null,
    "min": null,
    "show": true
  },
  "yBucketBound": "upper",
  "yBucketNumber": null,
  "yBucketSize": null
}
```

</p>
</details>

### CODE REVIEW

File: public/app/plugins/panel/heatmap/migrations.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ import { Options, defaultOptions, HeatmapColorMode } from './types';
 /** Called when the version number changes */
 export const heatmapMigrationHandler = (panel: PanelModel): Partial<Options> => {
   // Migrating from angular
-  if (Object.keys(panel.options).length === 0) {
+  if (panel.options === null || Object.keys(panel.options).length === 0) {
```
Comment (leeoniya) [inline public/app/plugins/panel/heatmap/migrations.ts:17]: maybe a bit shorter, (and also handles `undefined`):

```suggestion
  if (Object.keys(panel.options ?? {}).length === 0) {
```

Comment (overvenus) [reply public/app/plugins/panel/heatmap/migrations.ts]: Thanks, addressed in https://github.com/grafana/grafana/pull/79083/commits/7120465ee0fffbc87bad8d17776f2f1ece6c9387.

**Review by adela-almasan — APPROVED** [review]

File: public/app/plugins/panel/heatmap/migrations.test.ts [hunk 2]
```diff
@@ -91,6 +91,63 @@ describe('Heatmap Migrations', () => {
     `);
   });
 
+  it('Null Options', () => {
```
Comment (nmarrs) [inline public/app/plugins/panel/heatmap/migrations.test.ts:94]: great job including a test case!

**Review by nmarrs — APPROVED** [review]
🚀

**Pull request merged**